smallab/specification_generator.py

```diff
@@ -49,17 +51,28 @@ def generate(self, generation_specification: typing.Dict) -> typing.List[collect
             "a": [[1,2,3]]
         }
         '''
+
         generation_specification = collections.OrderedDict(sorted(generation_specification.items()))
         iterators = []
-        for key, value in generation_specification.items():
+
+        for index, key, value in enumerate(sorted(generation_specification.items())):
             if isinstance(value, list):
                 iterators.append(list(map(lambda x: (key, x), value)))
+
+                if value:  # in case the list is empty
+                    if not isinstance(value[0], (bool, int)):
+                        # One-hot encode values if not booleans or ints
+                        xs = [[val] for val in value]
+                        xs = np.array(xs)
+                        encoders.setdefault(index, OneHotEncoder().fit(xs))
```

Comment:
this shouldn't be done here, specification generator is not necessarily going to ever be called

Reply:
Oh wow, that's actually left from before sorry about that. The actual encoding happens in dashboard.py